src/classes/UTIL_Permissions.cls

```diff
@@ -167,6 +167,17 @@ public with sharing class UTIL_Permissions {
         return true;
     }
 
+    /**
+    * @description Instance method which can be mocked
+    * Determines whether the running user has delete permissions on the given object
+    *
+    * @param sObjectType SObjectType to check
+    * @return returns boolean to indicate whether the user has permission
+    */
+    public Boolean canDelete(Schema.SObjectType sObjectType) {
+        return canDelete(sObjectType.getDescribe().getName(), false);
```

Comment:
It looks like the other methods in this class call `UTIL_Describe.getObjectDescribe` when describing objects (which caches describe results).  Not totally necessary here IMO but would make this new method more consistent with the others in the same class if you want to align them.